pkg/handler-launcher-com/cmd/v1/cmd.proto

```diff
@@ -22,6 +22,7 @@ service Cmd {
   rpc GetFilesystems(EmptyRequest) returns (GuestFilesystemsResponse) {}
   rpc Ping(EmptyRequest) returns (Response) {}
   rpc Exec(ExecRequest) returns (ExecResponse) {}
+  rpc GuestPing(GuestPingRequest) returns (GuestPingResponse) {}
```

Comment:
As @vladikr pointed out in the community meeting, there should not be the need for a GRPC request at all here. This should be self-contained in the pod.

Reply:
using a rpc connection might reduce the memory footprint of the process being called in the pod exec. 

For example, calling a simple binary that can only make an rpc call to virt-launcher means we don't have to load in all the libvirt related libraries in that probe binary.

Reply:
Yes, this is already self-contained.
The reason we have the gRPC endpoint is because we want to talk to launcher.
The only request we might be able to do without is talking to the guest if that really already happens, but I think we're fine.

Reply:
Right now the flow is exactly the same as with exec probes here.
You specify it on the VMI, the controller puts in an exec probe on the pod, that calls virt-probe, that talks to launcher via gRPC and that then pings the guest-agent.
I don't see how much more self-contained it is.

The only improvement I could think about would be leveraging a readiness condition if we already have a guestAgent condition we could use. But we don't have that on the pod, so...

Reply:
Ok, so the grpc is only added to talk from within virt-launcher to virt-launcher. That makes sense. THanks.

/unhold

Reply:
On a quick check it looked to me like virt-handler would be involved.

Reply:
@rmohr @davidvossel final thoughts here for approval?